### PR TITLE
csproj.md: Emphasise implicit version context

### PR DESCRIPTION
**Pull request by dasMulli**
Emphasises that `RuntimeFrameworkVersion` and `NetStandardImplicitPackageVersion` only apply to .NET Core and .NET Standard projects respectively.

Context: https://github.com/NuGet/Home/issues/5809 - User expected `NetStandardImplicitPackageVersion` to work for a .NET Framework project.

### CODE REVIEW

File: docs/core/tools/csproj.md
Comment (mairaw) [inline docs/core/tools/csproj.md]: typo: tagreting
I'd also start with the condition: When targeting ..., never have ...

Also wondering if we want to be more explicit about the .NET Framework case here too?

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @dasMulli for following up on the issue and creating a PR for it. I think your changes greatly enhance the topic. I left a few comments to be addressed.

File: docs/core/tools/csproj.md
Comment (dasMulli) [reply docs/core/tools/csproj.md]: Thanks!

File: docs/core/tools/csproj.md
Comment (dasMulli) [inline docs/core/tools/csproj.md]: @mairaw I've added this bullet point on .NET Framework. I'm a little worried about updating versions of `NETStandard.Library` since 2.0.0 is bringing in targets and assemblies for ns2.0 support that might be dangerous if the user already has System.* packages installed in a packages.config project and no ns2.0 support installed.

Comment (nguerrera) [reply docs/core/tools/csproj.md]: @ericstj

**Review by nguerrera — APPROVED** [review]

**Comment by dasMulli** [comment]
Changed wording based feedback form @guardrex on slack

**Comment by dasMulli** [comment]
@mairaw could you take a look? any more changes necessary or ready to go?

File: docs/core/tools/csproj.md
Comment (mairaw) [inline docs/core/tools/csproj.md]: suggestion: install the needed version -> installs that version

**Review by mairaw — APPROVED** [review]
@dasMulli so sorry I dropped the ball on this one. I have a small suggestion. Let me know what you think before I merge this.

File: docs/core/tools/csproj.md
Comment (dasMulli) [reply docs/core/tools/csproj.md]: much better, changed it

Comment (mairaw) [reply docs/core/tools/csproj.md]: Cool. I'll merge when build finishes. Thanks @dasMulli and apologies again on the delay. Feel free to ping me when I'm taking more than a few days to respond.

**Pull request merged**
